internal/channelz/types_nonlinux.go

```diff
@@ -35,4 +31,6 @@ type SocketOptionData struct {
 // Getsockopt defines the function to get socket options requested by channelz.
 // It is to be passed to syscall.RawConn.Control().
 // Windows OS doesn't support Socket Option
-func (s *SocketOptionData) Getsockopt(fd uintptr) {}
+func (s *SocketOptionData) Getsockopt(fd uintptr) {
+	grpclog.Infof("Channelz: socket options are not supported on non-linux os and appengine.")
```

Comment:
How often will this be invoked?  We should wrap it in a `sync.Once` if it's frequent enough.

Reply:
It will be invoked when user try to get socket option for a socket on a platform that's not supported. I think log it once may make it easy to be overlooked. Or we can do `sync.Once(grpc.Warningf(...)` to only log once but draw more attention?